project-base/storefront/components/Pages/Order/TransportAndPayment/TransportAndPaymentSelect/TransportAndPaymentSelect.tsx

```diff
@@ -40,139 +37,21 @@ const TEST_IDENTIFIER = 'pages-order-';
-        const selectedTransport = transports.find((transport) => transport.uuid === selectedTransportUuid);
-
-        if (selectedTransport?.uuid === transport?.uuid) {
-            return;
-        }
-
-        if (!selectedTransport) {
-            await changeTransportInCart(null, null);
-            await handlePaymentChange(null);
-
-            return;
-        }
-
-        if (selectedTransport.isPersonalPickup || selectedTransport.transportType.code === 'packetery') {
-            if (!preSelectedPickupPlace) {
-                openPersonalPickupPopup(selectedTransport);
-
-                return;
-            }
-
-            await changeTransportInCart(selectedTransportUuid, preSelectedPickupPlace);
-            setPreSelectedPickupPlace(null);
-
-            return;
-        }
-
-        if (selectedTransportUuid !== transport?.uuid) {
-            await changeTransportInCart(selectedTransportUuid, null);
-        }
-    };
-
-    const handleGoPaySwiftChange = async (newGoPaySwiftValue: string | null) => {
-        await changePaymentInCart(payment?.uuid ?? null, newGoPaySwiftValue);
-    };
-
-    const loadPresetsFromLastOrder = async () => {
-        if (!transport) {
-            await handleTransportChange(lastOrderTransportUuid);
-        }
-        if (!payment) {
-            await handlePaymentChange(lastOrderPaymentUuid);
-        }
-    };
-
-    useEffect(() => {
-        loadPresetsFromLastOrder();
-    }, []);
-
-    const resetPaymentAndGoPayBankSwift = () => changePaymentInCart(null, null);
-
-    const resetAll = async () => {
-        await handleTransportChange(null);
-        await handlePaymentChange(null);
-        clearPacketeryPickupPoint();
-    };
-
-    const onChangePickupPlaceHandler = (selectedPickupPlace: ListedStoreFragmentApi | null) => {
-        if (selectedPickupPlace && preSelectedTransport) {
-            changeTransportInCart(preSelectedTransport.uuid, selectedPickupPlace);
-        } else {
-            handleTransportChange(null);
-            clearPacketeryPickupPoint();
-        }
-
-        setPreselectedTransport(null);
-    };
-
-    const onClosePickupPlacePopupHandler = () => {
-        clearPacketeryPickupPoint();
-        setPreselectedTransport(null);
-    };
-
-    const getPickupPlaceDetail = (transportItem: TransportWithAvailablePaymentsAndStoresFragmentApi) =>
-        transport?.uuid === transportItem.uuid &&
-        transportItem.stores?.edges?.some((storeEdge) => storeEdge?.node?.identifier === pickupPlace?.identifier)
-            ? pickupPlace
-            : null;
+    const { changePayment, changeGoPaySwift, resetPaymentAndGoPayBankSwift } =
+        usePaymentChangeInSelect(changePaymentInCart);
+    const { preSelectedTransport, changeTransport, changePickupPlace, closePickupPlacePopup } =
+        useTransportChangeInSelect(transports, lastOrderPickupPlace, changeTransportInCart, {
+            onTransportReset: async () => changePayment(null),
```

Comment:
I am not sure what you mean. It is true that keeping or removing async is the same, but I left the async to be more clear and transparent about it being an asynchronous operation.

If you check the type of the callback:
<img width="368" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/073570a4-c855-46ce-870a-189189be2088">

Then you can see that this is exactly the type of the provided function:
<img width="442" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/43fa4c6c-a2ed-4e56-a628-5d6c052d5053">


The reason, why it is not awaited here, is because of IoC, as the handler is responsible for dispatching and handling the callbacks. From the outside, I can only adhere to the API.

This function is therefore provided, called, awaited:
<img width="331" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/3ed35854-eba6-4420-a347-58e390941a38">